util/serial/src/lib.rs

```diff
@@ -1,7 +1,5 @@
 // Copyright (c) 2018-2022 The MobileCoin Foundation
 
-#![no_std]
```

Comment:
This was done to allow the surfaced `std::num::TryFromIntError` being used when attempting to convert from u128 to u64 type integers. I don't have the original context for why this line was originally needed. Is there a workaround or alternative error type that I should be using here?

Reply:
This cannot be removed or we will break the enclave build. Lets see why we wanted to remove this

Reply:
It looks like some sort of definitions conflict in our crates elsewhere. From the logs:
```
error: duplicate lang item in crate `std` (which `mc_util_serial` depends on): `panic_impl`.
    |
    = note: the lang item is first defined in crate `mc_sgx_panic` (which `mc_sgx_compat` depends on)
    = note: first definition in `mc_sgx_panic` loaded from /__w/mobilecoin/mobilecoin/target/consensus-enclave/x86_64-unknown-linux-gnu/debug/deps/libmc_sgx_panic-52e2d6de1dea2756.rmeta
    = note: second definition in `std` loaded from /opt/rustup/toolchains/nightly-2022-04-29-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-202fc93d8ccaebf2.rlib

  error: duplicate lang item in crate `std` (which `mc_util_serial` depends on): `oom`.
    |
    = note: the lang item is first defined in crate `mc_sgx_alloc` (which `mc_sgx_compat` depends on)
    = note: first definition in `mc_sgx_alloc` loaded from /__w/mobilecoin/mobilecoin/target/consensus-enclave/x86_64-unknown-linux-gnu/debug/deps/libmc_sgx_alloc-32f75c292bdb214a.rmeta
    = note: second definition in `std` loaded from /opt/rustup/toolchains/nightly-2022-04-29-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-202fc93d8ccaebf2.rlib

  error: duplicate lang item in crate `panic_unwind` (which `std` depends on): `eh_personality`.
    |
    = note: the lang item is first defined in crate `mc_sgx_panic` (which `mc_sgx_compat` depends on)
    = note: first definition in `mc_sgx_panic` loaded from /__w/mobilecoin/mobilecoin/target/consensus-enclave/x86_64-unknown-linux-gnu/debug/deps/libmc_sgx_panic-52e2d6de1dea2756.rmeta
    = note: second definition in `panic_unwind` loaded from /opt/rustup/toolchains/nightly-2022-04-29-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-1333333cbe389678.rlib

  error: could not compile `mc-sgx-report-cache-api` due to 3 previous errors
```

Reply:
using the `core` crate instead of the `std` crate works and fixes the build